File: Crio/Recursion/8_PermutationPalindrome/PermutationPalindrome.java
import java.io.*;
import java.util.*;
import java.lang.Math;
public class  PermutationPalindrome {

  public static int isPermutationPalindrome(String s ) {
    // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    int size = s.length();
    int oddCount = 0;
    int i = 0, j = 0;
    HashMap< Character , Integer> count = new HashMap<Character, Integer> () ;

    while (i < size ) {
      char x = s.charAt(i);
      count.put(x, count.getOrDefault(x, 0) + 1);
      i++;
    }
    for ( char it : count.keySet()) {
      int x = count.get(it);
      if ( x % 2 == 1) oddCount++;
    }
    if ( oddCount > 1) return 0;
    return 1;
    // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
  }
  public static void  main (String args []) {

    Scanner sc =  new  Scanner (System.in);
    int t = 1;
    t = sc.nextInt();
    sc.nextLine();
    while (t > 0) {
      t--;
      String s = new String();
      s = sc.next();

      int flag = isPermutationPalindrome(s);
      if (flag == 1) {
        System.out.println("True");
      } else {
        System.out.println("False");
      }
    }
  }

}

File: Crio/Recursion/8_PermutationPalindrome/PermutationPalindrome.cpp
#include<bits/stdc++.h>
using namespace std;

bool isPermutationPalindrome(string &s) {
	// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
	int count[55];
	memset(count , 0 , sizeof count);
	for(int i = 0 ; i < s.size() ; i++) {
		if(s[i] <= 'z' && s[i] >= 'a')
			count[s[i] - 'a']++;
	 	if(s[i] <= 'Z' && s[i] >= 'A')
			count[26 + s[i] - 'A']++;
	 	
	}
	int oddCount = 0;
	for(int i = 0 ;i < 55 ; i++) {
		if(count[i]%2 != 0) {
			oddCount++;
		}
	}
	if(oddCount > 1) {
		return false;
	}
	return true;
	// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
}

int main()
{
	int t;
	cin >> t;
	while(t--) {
		string s;
		cin >> s;
		bool answer = isPermutationPalindrome(s);
		if(answer) {
			cout << "True\n";
		} else {
			cout << "False\n";
		}
	}
	return 0;
}
